pkg/errors/errors.go

```diff
@@ -0,0 +1,50 @@
+package errors
+
+import (
+	"errors"
+
+	"github.com/contiamo/go-base/pkg/models"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+var (
+	// ErrPermission is the standard "Permission Denied" error
+	ErrPermission = errors.New("You don't have required permission to perform this action")
+	// ErrAuthorization is the standard "Unauthorized" error
+	ErrAuthorization = errors.New("User is unauthorized, make sure you've logged in")
+	// ErrInternal is the standard "Internal Server" error
+	ErrInternal = errors.New("Internal server error, please try again later")
+	// ErrInvalidParameters is the standard "Bad Request" error
+	ErrInvalidParameters = errors.New("Some of the request parameters are not correct")
+	// ErrUnmarshalling is the JSON deserialization error
+	ErrUnmarshalling = errors.New("Failed to read JSON from the request body")
+	// ErrForm is the form parsing error
+	ErrForm = errors.New("Failed to parse the submitted form")
+	// ErrNotFound is the standard entiry not found error
+	ErrNotFound = errors.New("The requested object was not found")
+	// ErrNotImplemented is intended to be used when stubbing new endpoints
+	ErrNotImplemented = errors.New("Method is not implemented")
+)
+
+// ValidationErrors contains errors organized by validated fields
+// for now it's just an alias to the validation library we use
+type ValidationErrors = validation.Errors
+
+// ValidationErrorsToFieldErrorResponse converts validation errors to the format that is
+// served by HTTP handlers
+func ValidationErrorsToFieldErrorResponse(errs ValidationErrors) (fieldErrResp models.FieldErrorResponse) {
```

Comment:
Is there anyway for it to allow both `ValidationErrors` and `validation.Errors`?

Reply:
I think we should not allow that. We mask the library type with our own in case we would like to switch the library and it would not affect the http handlers.

I personally don't know how to allow both except using `interface{}` and casting to known types.